comms/src/connection/dealer_proxy.rs

```diff
@@ -104,13 +113,13 @@ impl DealerProxy {
 
     /// Proxy the source and sink addresses. This method does not block and returns stores the [thread::JoinHandle] in
     /// the DealerProxy.
-    pub fn spawn_proxy(&mut self) {
-        self.thread_handle = Some(spawn_proxy(
+    pub fn spawn_proxy(&mut self) -> Result<()> {
```

Comment:
you are not sending though an empty result Ok(()) here, you are sending a result Ok(self.thread_handle). Looking at the code, it looks like you only want to send back Ok(()). If this is the case you need to have a separate Ok(()) at the end of the function.

Reply:
Issue was fixed by a different PR.